source/core/slang-char-encode.cpp

```diff
@@ -135,25 +135,6 @@ class Utf16CharEncoding : public CharEncoding //UTF16
             return CharEncodeType::UTF16Reversed;
         }
     }
-    else
```

Comment:
I think we still want to keep this logic, but to make sure we don't treat trailing null terminator as utf16 string.

Reply:
Are you sure?  This code was only used if bytesCount == 0 or 1.  There's not enough data to guess UTF16.  What am I missing?

Reply:
If we keep the UTF16 guessing logic, it will have to be inside the bytesCount >= 2 block above because it makes no sense to only guess UTF16 on strings with length 0 and 1.

Reply:
Yes, you are right. The guessing logic should be in the >=2 block instead of in this branch.